feat(Pagination): add Pagination component

mking-clari opened the pull request:
References #77

* Add `Pagination`, `PaginationItem`, `PaginationLink`
* `Pagination` supports sizing
* `PaginationItem` supports active and disabled states
* `PaginationLink` supports special rendering for next and previous links

eddywashere commented on src/PaginationLink.jsx:
```diff
@@ -0,0 +1,43 @@
+  );
+
+  return (
+    <Tag {...attributes} className={classes}>
```
Can you add the aria label, the aria hidden attr and sr-only label for accessibility?

```
<a class="page-link" href="#" aria-label="Next">
    <span aria-hidden="true">&raquo;</span>
    <span class="sr-only">Next</span>
</a>
```

eddywashere replied:
For prev/next, I don't think it will ever require children unless someone wants to override the carets. Could you setup rendering to be: if prev/next, include spans and render children instead of carets vs rendering children before the prev/next content.

eddywashere commented:
A few comments, let me know what you think. Great work!

mking-clari commented:
@eddywashere Thanks for the feedback, I updated the code.

Some notes:
* If previous or next is set, the children prop replaces just the caret, not all the spans. If the user wants full customization of the spans, they can just not set previous or next.
* The `aria-label` prop is used for both the `aria-label` attribute and the `.sr-only` text.

eddywashere commented on src/PaginationLink.jsx:
```diff
@@ -0,0 +1,67 @@
+
+  let defaultCaret;
+  if (previous) {
+    defaultCaret = <span>&laquo;</span>;
```
I don't think these extra spans are needed anymore since it's taken care of with `<span aria-hidden="true">{children || defaultCaret}</span>`.

eddywashere replied:
I don't remember if `&laquo;` as a string will render html, might need html entity.

eddywashere commented on src/PaginationLink.jsx:
```diff
@@ -0,0 +1,67 @@
+      aria-label={ariaLabel}
+    >
+      {(previous || next) ?
+        (<span>
```
extra span here too. you could push the child content to an array to get around the need for the wrapper.

mking-clari commented:
@eddywashere For the `&laquo;` issue, I replaced the html entity with the unicode character (`\u00ab`), which should [avoid the escaping issue](https://facebook.github.io/react/docs/jsx-gotchas.html). Also updated the test for consistency.

For the span issue, I added an array (keys required to avoid warning).

eddywashere commented:
👍
